src/main/java/org/cactoos/list/ListIterator.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.list;
+
+/**
+ * Iterator of the list.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <T> Items type
+ * @since 0.34
+ */
+public final class ListIterator<T> implements java.util.ListIterator<T> {
+
+    /**
+     * Original list iterator.
+     */
+    private final java.util.ListIterator<T> origin;
+
+    /**
+     * Ctor.
+     * @param orig Original list iterator.
+     */
+    public ListIterator(final java.util.ListIterator<T> orig) {
+        this.origin = orig;
+    }
+
```

Comment:
@proshin-roman I think it would be interesting to have two other constructors  `ListIterator(List<>, int)` and  `ListIterator(List<>)` so we can extract an immutable `ListIterator` (with its index set) directly from a `List` whenever we need one, what do you think?

Reply:
@paulodamaso Ok, I've tried to add such ctors. So they will look like:
```java
public ListIterator(final List<T> list) {
    this(list.listIterator());
}

public ListIterator(final List<T> list, final int index) {
    this(list.listIterator(index));
}
```
and there is a problem that both these ctors violate the second principle of Cactoos (see "No code in constructors" [here](https://www.elegantobjects.org/#principles)).
It means that either these ctors have no place in the class or they should be refactored so that the code is lazily executed. But the second option would need additional refactoring of the class. WDYT?

Reply:
@proshin-roman See if refactoring adds too much effort to the issue; if yes, just let a `@todo` here so we can do this in the future.